components/expression/expression-renderer.js

```diff
@@ -172,21 +211,66 @@ genotet.ExpressionRenderer.prototype.drawExpressionMatrix_ = function() {
  * @private
  */
 genotet.ExpressionRenderer.prototype.drawMatrixCells_ = function() {
-  // TODO(liana): Implement this...
-  var cells = this.svgHeatmapContent_.selectAll('rect.cell').data([{
-    width: this.heatmapWidth_,
-    height: this.heatmapHeight_
-  }]);
-  cells.enter().append('rect')
-    .classed('cell', true);
+  var heatmapData = this.data.matrix;
+  var cellWidth = this.heatmapWidth_ / heatmapData.conditionNames.length;
+  var cellHeight = this.heatmapHeight_ / heatmapData.geneNames.length;
+  //console.log(heatmapData);
+  var colorScale = d3.scale.linear()
+    .domain([heatmapData.valueMin, heatmapData.valueMax])
+    .range(genotet.data.redYellowScale);
+
+  this.svgHeatmapContent_.selectAll('g').remove();
```

Comment:
Why removing all 'g'? This means you want to re-render everything again, and this should be avoided.

For example, when the view is resized, you only need to adjust the cell sizes. But if you remove all g's, then every element has to be added to the HTML again, which is costly. By doing so, you miss the purpose of d3.selectAll().data().